dot/core/service.go

```diff
@@ -380,7 +382,7 @@ func (s *Service) ProcessBlockAnnounceMessage(msg network.Message) error {
 		blockRequest := &network.BlockRequestMessage{
 			ID:            randomID, // random
 			RequestedData: 2,        // block body
-			StartingBlock: append([]byte{0}, currentHash[:]...),
+			StartingBlock: append([]byte{1}, s.blockState.LatestHeader().Number.Bytes()...),
```

Comment:
the number needs to be in 8-byte little endian format, so convert the number to int64 -> uint64 -> bytes by doing
```
buf := make([]byte, 8)
binary.LittleEndian.PutUint64(buf, uint64(number.Int64()))
```

why can't we use hash here?